common/priorityQueue.py

```diff
@@ -0,0 +1,84 @@
+import bisect
+
+class PriorityQueue:
+    def __init__(self):
+        self._queue = []
+        self._size = 0
+        self._id = 0
+
+        # a unique sender is determined by the path to the sender's emails
+        self._uniqueSenders = set([])
+
+    def getLength(self):
+        return self._size
+
+    # returns the popped item
+    # returns None if no item is popped off
+    def push(self, item, priority):
+        # maintains the queue to have 10 elements or less
+        # element with lowest priority gets popped off
+        # in this case, the element with the lowest probability gets popped off
+
+        # checks to see if this sender is already in that priority queue
+        # if so, checks to see if probability should be updated
+        thisSender = item[0]
+        if thisSender in self._uniqueSenders:
+            senderIndex = 0
```

Comment:
Minor: None would be a better default value than 0 (and if the subsequent code can't find a match, it's better to throw an error than to use the wrong value, as the error presumably indicates some bug in the code).